src/server_manager/cloud/gcp_api.ts

```diff
@@ -0,0 +1,437 @@
+   *   documentation for naming requirements.
+   * @param zoneId - The zone in which to create the instance.
+   * @param installScript - A script to run once the instance has launched.
+   */
+  createInstance(
+      projectId: string, name: string, zoneId: string, size: string,
+      installScript: string): Promise<Operation> {
+    const data = {
+      name,
+      machineType: `zones/${zoneId}/machineTypes/${size}`,
+      disks: [
+        {
+          boot: true,
+          initializeParams: {
+            sourceImage: 'projects/ubuntu-os-cloud/global/images/family/ubuntu-1804-lts',
+          },
+        },
+      ],
+      networkInterfaces: [
+        {
+          network: 'global/networks/default',
+          // Empty accessConfigs necessary to allocate ephemeral IP
+          accessConfigs: [{}],
+        },
+      ],
+      serviceAccounts: [
+        {
+          scopes: [
+            'https://www.googleapis.com/auth/userinfo.email',
+            'https://www.googleapis.com/auth/compute.readonly',
+            'https://www.googleapis.com/auth/devstorage.read_only',
+          ],
+        },
+      ],
+      labels: {
+          // `${label}`: true,
+      },
+      tags: {
+        items: [name],
+      },
+      metadata: {
+        items: [
+          {
+            key: 'enable-guest-attributes',
+            value: 'TRUE',
+          },
+          {
+            key: 'user-data',
+            value: installScript,
+          },
+        ],
+      },
+    };
+    // TODO: Figure out how to do this in the object itself.
+    // @ts-ignore
+    data.labels[label] = 'true';
+    return this.computeHttpClient.post<Operation>(
+        `compute/v1/projects/${projectId}/zones/${zoneId}/instances`,
+        data,
+    );
+  }
+
+  /**
+   * Deletes a specified Google Compute Engine VM instance.
+   *
+   * @see https://cloud.google.com/compute/docs/reference/rest/v1/instances/delete
+   *
+   * @param projectId - The GCP project ID.
+   * @param instanceId - The ID of the instance to delete.
+   * @param zoneId - The zone in which the instance resides.
+   */
+  deleteInstance(projectId: string, instanceId: string, zoneId: string): Promise<Operation> {
+    return this.computeHttpClient.delete<Operation>(
+        `compute/v1/projects/${projectId}/zones/${zoneId}/instances/${instanceId}`,
+    );
+  }
+
+  /**
+   * Gets the specified Google Compute Engine VM instance resource.
+   *
+   * @see https://cloud.google.com/compute/docs/reference/rest/v1/instances/get
+   *
+   * @param projectId - The GCP project ID.
+   * @param instanceId - The ID of the instance to retrieve.
+   * @param zoneId - The zone in which the instance resides.
+   */
+  getInstance(projectId: string, instanceId: string, zoneId: string): Promise<Instance> {
+    return this.computeHttpClient.get<Instance>(
+        `compute/v1/projects/${projectId}/zones/${zoneId}/instances/${instanceId}`,
```

Comment:
All the URL building needs proper escaping. You should treat the inputs as not url safe.

Notice how we pass a dictionary of parameters for DigitalOcean, which makes it easier. But you also need to safely build the path.